compiler/rustc_mir/src/borrow_check/type_check/mod.rs

```diff
@@ -180,7 +179,66 @@ pub(crate) fn type_check<'mir, 'tcx>(
             liveness::generate(&mut cx, body, elements, flow_inits, move_data, location_table);
 
             translate_outlives_facts(&mut cx);
-            cx.opaque_type_values
+            let mut opaque_type_values = cx.opaque_type_values;
+
+            for (_, revealed_ty) in &mut opaque_type_values {
```

Comment:
Why does `opaque_type_values` contain elements at the start of the loop? It seems to me that the only place where elements were inserted into this field was in the code you deleted below?

Reply:
yes, but this function is called multiple times. The opaque types are collected over time.